org.eclipse.xtext/src/org/eclipse/xtext/EcoreUtil2.java

```diff
@@ -189,6 +189,10 @@ public static <T extends EObject> List<T> getAllContentsOfType(EObject ele, Clas
 		return Lists.newArrayList(Iterators.filter(ele.eAllContents(), type));
 	}
 
+	public static <T extends EObject> List<T> getAllContentsOfType(Resource resource, Class<T> type) {
```

Comment:
am not sure if we want this. @szarnekow @LorenzoBettini wdyt

@tfonda-fbk can you please add a `@since 2.37`

Reply:
Not a fan but I've also no hard feelings about the new method.